packages/fw-avatar/test/fw-avatar-group.test.js

```diff
@@ -0,0 +1,247 @@
+    await element.elementUpdated;
+
+    const itemsListHtml = element.shadowRoot.querySelectorAll(
+      "mwc-list-item.items-list-item > span"
+    );
+
+    let tempArray = Array();
+    for (let i = 0; i < itemsListHtml.length; i = i + 1) {
+      tempArray.push({
+        name: String(itemsListHtml[i].textContent.trim()),
+      });
+    }
+    expect(element.items).to.equal(usersWithOutSecondaryAttribute);
+    expect(tempArray).to.be.eql(usersWithOutSecondaryAttribute);
+  });
+
+  it("checks Max Count property", async () => {
+    const visibleAvatars = element.shadowRoot
+      .querySelector("#group")
+      .querySelectorAll("fw-avatar[type='initials']");
+    expect(element.maxCount).to.equal(maxCount);
+    expect(visibleAvatars.length).to.equal(
+      Math.min(users.length, maxCount),
+      "The number of avatars rendered in the group do not match with MaxCount\n"
+    );
+  });
+
+  it("checks menu visiblility on tap/Click event", async () => {
+    expect(
+      element.shadowRoot.querySelector("#menu").getAttributeNames()
+    ).not.to.include("open", "Menu in open mode without Click event \n");
+
+    const avatarGroupButton = element.shadowRoot.querySelector(".group");
+
+    setTimeout(() => {
+      avatarGroupButton.click();
+    });
+    const {} = await oneEvent(avatarGroupButton, "click");
+
+    expect(
+      element.shadowRoot.querySelector("#menu").getAttributeNames()
+    ).to.include("open", "Click even didn't triggered the menu to  open \n");
+  });
+
+  it("checks Header property", async () => {
+    let headertext = element.shadowRoot
+      .querySelector(".dialog-header-span")
+      .textContent.trim();
+
+    expect(element.header).to.equal(header);
+    expect(headertext).to.be.equal(
+      header,
+      "Header do not match with given prop\n"
+    );
+    element.removeAttribute("header");
+    await element.elementUpdated;
+    headertext = element.shadowRoot
+    .querySelector(".dialog-header-span")
+    .textContent.trim();
+
+    expect(headertext).to.equal("Members","Default Header Value do not match");
+
+  });
+
+  it("checks for searchBar visibility", async () => {
+    const searchBarElement = element.shadowRoot.querySelector("#searchBar");
+
+    if (!showSearchBar)
+      expect(
+        searchBarElement,
+        "Search Bar visible , even when false passed to showSearchBar\n"
+      ).to.be.null;
+    else
+      expect(
+        searchBarElement,
+        "Search Bar not visible when true passed to showSearchBar\n"
+      ).not.to.be.null;
+  });
+
+  it("checks for primary action property", async () => {
+    const primaryActionItem = element.shadowRoot.querySelector("#primary-action-item");
+
+    expect(primaryActionItem.querySelector("span").textContent).to.equal(
+      primaryAction["title"]
+    );
+    expect(primaryActionItem.querySelector("mwc-icon")).dom.to.equal(
+      `<mwc-icon>${primaryAction["icon"]}</mwc-icon>`,
+      { ignoreAttributes: ["slot"] }
```

Comment:
Why ignore the slot attribute?

Reply:
Test focused on checking if the DOM consist of mwc-icon of the icon provided by the primaryAction object's icon property. Value of slot was a static value in the component, so it would be the same for all created elements. Thats why i only checked for presence of mwc-icon with primaryActions icon property value.

Reply:
Got it